File: #203_Remove_Linked_List_Elements/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if (head == null) return null;
        while (head != null && head.val == val) head = head.next;
        if (head == null || head.next == null) return head;
        ListNode prev = head;
        ListNode cur = head.next;
        while (cur != null) {
        	if (cur.val == val) {
        		prev.next = cur.next;
        		cur = cur.next;
        	}
        	else {
        		prev = prev.next;
        		cur = cur.next;
        	}
        }
        return head;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	ListNode head = new ListNode(2);
    	ListNode node1 = new ListNode(2);
    	ListNode node2 = new ListNode(2);
    	ListNode node3 = new ListNode(2);
    	head.next = node1;
    	node1.next = node2;
    	node2.next = node3;
    	ListNode.printList(head);
    	ListNode.printList(sol.removeElements(head, 2));
    }
}

File: #203_Remove_Linked_List_Elements/Solution.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def removeElements(self, head, val):
    	dummy = ListNode(0)
    	dummy.next = head
    	prev = dummy
    	cur = head
    	while cur:
    		if cur.val == val:
    			tmp = cur.next
    			cur.next = None
    			cur = tmp
    			prev.next = cur
    		else:
    			prev = prev.next
    			cur = cur.next
    	return dummy.next

sol = Solution()
head = ListNode(1)
n1 = ListNode(2)
n2 = ListNode(4)
n3 = ListNode(10)
head.next = n1
n1.next = n2
n2.next = n3
print sol.removeElements(head, 9).val